pkg/grpc/authn/basic/extractor.go

```diff
@@ -23,9 +24,23 @@ const (
 )
 
 var (
-	log = logging.NewRateLimitLogger(logging.LoggerForModule(), cacheSize, 1, rateLimitFrequency, logBurstSize)
+	once sync.Once
+	log  *logging.RateLimitedLogger
 )
 
+func getRateLimitedLogger() *logging.RateLimitedLogger {
```

Comment:
why not doing it in init? Then we don't need `sync.Once`

Reply:
https://github.com/stackrox/stackrox/actions/runs/4989382445
The `TestAuditLogCollectionManager: github.com/stackrox/rox/sensor/common/compliance` and `TestNodeInventoryHandler: github.com/stackrox/rox/sensor/common/compliance` tests fail multiple times with:
```
    node_inventory_handler_test.go:53: found unexpected goroutines:
        [Goroutine 28 in state select, with github.com/stackrox/rox/pkg/logging.(*RateLimitedLogger).logFlushLoop on top of the stack:
        goroutine 28 [select]:
        github.com/stackrox/rox/pkg/logging.(*RateLimitedLogger).logFlushLoop(0xc0009fdc40)
        	/__w/stackrox/stackrox/pkg/logging/rate_limited_logger.go:176 +0x24a
        created by github.com/stackrox/rox/pkg/logging.NewRateLimitLogger
        	/__w/stackrox/stackrox/pkg/logging/rate_limited_logger.go:61 +0x631
    ]
```